### DISCUSSION TITLE
How to order by aggregate field?

### DISCUSSION THREAD

**Discussion by stefan-girlich** [supabase/supabase#26376] [Questions]
Let's say I have tables `articles` and tables `purchases`. `purchases` references `articles` via foreign key and depicts how often an article was included in a single purchase (`article_count`).

The following query lists how often a given customer has purchased each article:

```ts
const { data } = await supabase
        .from('articles')
        .select('*, purchases!inner(customer_id, article_count.sum())')
        .eq('purchases.customer_id', customerId)
```

Result:
```js
{
    // ...
    "purchases": [
      {
        "sum": 10,
        "customer_id": "foo"
      }
    ]
```

Is there a way to order by `sum` via `.orderBy()`, similar to how [ordering by nested entities](https://supabase.com/blog/postgrest-11-prerelease#order-by-related-tables) works?

**Comment by sanjaybaskaran01** [comment]
Would like to know how this is possible too!

**Comment by redshift75** [comment]
Having the same issue - when I use an aggregate column alias in order clause I get an error that the column is not found
